### ISSUE TITLE
DSTU3: Extensions in Reference objects work in 1.4 but not in 1.5

### ISSUE THREAD

**Issue by CarthageKing** [jamesagnew/hapi-fhir#377]
Hi,

We are adding the http://hl7.org/fhir/StructureDefinition/encounter-primaryDiagnosis extension to either an Encounter.hospitalization.admittingDiagnosis or Encounter.hospitalization.dischargeDiagnosis to indicate that a certain Condition link is the primary diagnosis.

This works well in HAPI FHIR DSTU3 1.4 but it now seems to be broken in 1.5. When using 1.5, we are getting the following:
```java
Exception in thread "main" java.lang.NullPointerException
	at ca.uhn.fhir.util.FhirTerser.cloneInto(FhirTerser.java:151)
	at ca.uhn.fhir.parser.BaseParser.preProcessValues(BaseParser.java:716)
	at ca.uhn.fhir.parser.XmlParser.encodeCompositeElementToStreamWriter(XmlParser.java:642)
	at ca.uhn.fhir.parser.XmlParser.encodeChildElementToStreamWriter(XmlParser.java:544)
	at ca.uhn.fhir.parser.XmlParser.encodeCompositeElementToStreamWriter(XmlParser.java:696)
	at ca.uhn.fhir.parser.XmlParser.encodeResourceToXmlStreamWriter(XmlParser.java:785)
	at ca.uhn.fhir.parser.XmlParser.encodeResourceToXmlStreamWriter(XmlParser.java:759)
	at ca.uhn.fhir.parser.XmlParser.encodeChildElementToStreamWriter(XmlParser.java:566)
	at ca.uhn.fhir.parser.XmlParser.encodeCompositeElementToStreamWriter(XmlParser.java:696)
	at ca.uhn.fhir.parser.XmlParser.encodeChildElementToStreamWriter(XmlParser.java:544)
	at ca.uhn.fhir.parser.XmlParser.encodeCompositeElementToStreamWriter(XmlParser.java:696)
	at ca.uhn.fhir.parser.XmlParser.encodeResourceToXmlStreamWriter(XmlParser.java:785)
	at ca.uhn.fhir.parser.XmlParser.encodeResourceToXmlStreamWriter(XmlParser.java:759)
	at ca.uhn.fhir.parser.XmlParser.doEncodeResourceToWriter(XmlParser.java:177)
	at ca.uhn.fhir.parser.BaseParser.encodeResourceToWriter(BaseParser.java:380)
	at ca.uhn.fhir.parser.BaseParser.encodeResourceToString(BaseParser.java:364)
	at TestPrimaryDiagnosisExtension.main(TestPrimaryDiagnosisExtension.java:47)
```

I'm attaching a simple Maven project that reproduces the problem.
[hapi-fhir-primarydiagnosis-extension-test.zip](https://github.com/jamesagnew/hapi-fhir/files/294636/hapi-fhir-primarydiagnosis-extension-test.zip)

**Comment by CarthageKing** [comment]
Just tried this against master and the issue seems to be fixed there.

**Comment by jamesagnew** [comment]
Thanks for reporting!

Yup, this was fixed in a recent checkin. 1.6 snapshots are available, and we'll be releasing 1.6 soon.
